network/src/protocols/discovery/addr.rs

```diff
@@ -52,50 +46,36 @@ pub trait AddressManager {
 
 // bitcoin: bloom.h, bloom.cpp => CRollingBloomFilter
 pub struct AddrKnown {
-    max_known: usize,
-    addrs: HashSet<RawAddr>,
-    addr_times: HashMap<RawAddr, Instant>,
-    time_addrs: BTreeMap<Instant, RawAddr>,
+    filters: StableBloomFilter<DefaultBuildHashKernels<RandomState>>,
 }
 
 impl AddrKnown {
     pub(crate) fn new(max_known: usize) -> AddrKnown {
         AddrKnown {
-            max_known,
-            addrs: HashSet::default(),
-            addr_times: HashMap::default(),
-            time_addrs: BTreeMap::default(),
+            filters: StableBloomFilter::new(
```

Comment:
Please explain why choosing these parameters

Reply:
Please refer to http://webdocs.cs.ualberta.ca/~drafiei/papers/DupDet06Sigmod.pdf chapter 4.1 and explain how the parameters are set.

This is why I suggest to create a new PR for bloom filter.

Reply:
Memory usage = m * d = 5000 * 3 bytes
Verifiable data volume = m / d = 5000 / 3 =  1600+